Migrate: task-status-pie-chart

lachlan-robinson opened the pull request:
# Description

Migrates the task status pie chart to ngx charts used in 9.x

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [x] Before and after comparison.
- [x] Change target grade and ensure data is correctly displayed.
- [x] Check different units to ensure data is correctly displayed.

### Before

![Screenshot 2025-04-10 185650](https://github.com/user-attachments/assets/06cabcea-9d15-42dc-b049-cc704acf3da1)

### After

![Screenshot 2025-04-10 185708](https://github.com/user-attachments/assets/7a17e88b-b6ce-4cbc-9e6d-0204dc59a932)

## Testing Checklist:

- [x] Tested in latest Chrome
- [ ] Tested in latest Safari
- [x] Tested in latest Firefox

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have requested a review from @macite and @jakerenzella on the Pull Request

zara1504 committed:
chore: remove media-service.coffee and modify doubtfire-angularjs.module.ts

HasinduWelarathne committed:
Deleting debug.coffee

lachlan-robinson committed:
chore: migrate pie chart to ngxchart

SahiruWithanage reviewed (approved):
Hey, great job on the  migration! The code is clean and organized, and during local testing the hover effects and legend highlighting performed exactly as expected. You have adhered to TypeScript and Angular best practices, and the integration with the legacy AngularJS module remains seamless. Overall, it’s a solid improvement that modernizes our codebase without disrupting any existing functionality.

<img width="1008" alt="Screenshot 2025-04-18 at 6 22 29 am" src="https://github.com/user-attachments/assets/d0df48f6-b7ba-4990-af93-f7d57ede4bfe" />

disururathnayake commented:
I ran it locally and everything functioned as expected. You've clearly applied Angular and TypeScript standards effectively.  Nice work on the migration! One small suggestion: adding a few comments in the **taskstatuspiechart.component.ts** file could help improve readability and make it easier for others to follow the logic. Overall, great job!

atharv02-git reviewed (approved):
Hi @lachlan-robinson, was able to pull your work and Just went through it, below are my findings:

- [x] `Task-Status-Pie-Chart-Component` has been imported and declared inside the necessary angular.module.ts.
- [x] `Task-Status-Pie-Chart` has been properly unlinked and downgraded from angularjs.module.ts and migrated `Task-Status-Pie-Chart-Component` has been linked.
- [x] `Task-Status-Pie-Chart` has been properly referenced in all the other necessary components like `project-progress-dashboard.component.html` and `project-dashboard.tpl.html` along with the updated property binding style to use the Angular form.
- [x] Old Coffee file has been sucessfully replaced with it's corresponding `.ts` file.
- [x] ngx charts have been properly imported into the component's html file.
- [x] Was able to build and then run the change without any errors.
- [x] Component along with the legend (on hover) looks even more better and stylish after the migration.
![image](https://github.com/user-attachments/assets/9dd99d2a-e9a0-4db3-987e-36bb3179e9f5)


Just a minor change in the below code; you can get rid of the `console.log(event)`. While it is a good practice when you're testing but on the production side, it's better to hide events. Pretty much happy with rest of the things!
![image](https://github.com/user-attachments/assets/47e1ab39-7af0-4d05-aed1-a3ba1b954e52)

Good Job with the migration again! LGTM! 
Good Luck now with the migration for `progress-dashboard` component. Can't wait to see you close the entire loop!

lachlan-robinson committed:
chore: remove testing console log

lachlan-robinson committed:
chore: remove testing console log

lachlan-robinson committed:
chore: restore package and lock to 9.x

b0ink committed:
fix: support building on windows

b0ink committed:
chore: update package-lock

aNebula commented on src/app/visualisations/task-status-pie-chart/taskstatuspiechart.component.html:
```diff
@@ -0,0 +1,12 @@
+    [customColors]="colors"
+    [legend]="true"
+    [legendTitle]="''"
+    [legendPosition]="'right'"
```
The legends seem to be in a different place than before. Does `legendPosition` need to be changed to show on the top?

lachlan-robinson replied:
ngx-chart does not natively support the legend being placed above. To maintain consistency I chose right to match the migrated burndown-chart present in 9.x

aNebula replied:
I see. Okay, that's alright then.

aNebula commented on src/app/visualisations/task-status-pie-chart/taskstatuspiechart.component.ts:
```diff
@@ -0,0 +1,87 @@
+    }
+  }
+
+  /**
+   *
+   * Hover event handler for the pie chart segments.
+   *
+   */
+
+  // onSelect(event) {
+  //
+  //   console.log(event.value);
+  //}
```
Please do not leave testing codes as comments.

lachlan-robinson replied:
resolved.

aNebula reviewed (changes requested):
Please make the changes and take it straight through leadership review.

lachlan-robinson committed:
chore: remove testing comments

aNebula reviewed (approved):
LGTM!
@lachlan-robinson Please open a upstream PR against [doubtfire-lms/doubtfire-web 9.x branch](https://github.com/doubtfire-lms/doubtfire-web/tree/9.x).

aditya993388 committed:
remove legacy CoffeeScript files and references

macite committed:
Merge pull request #940 from HasinduWelarathne/9.x

Remove/Unused debug.coffee module and related references

macite committed:
Merge pull request #941 from aditya993388/feature/remove-legacy-create-unit-modal-files

Remove legacy files and references(Completing the create-unit-modal migration

macite committed:
Merge pull request #925 from b0ink/fix/9.x-windows-build

fix: support building on windows [9.x]

returnMarcco committed:
Migrate/group set selector (#981)

* chore: migrate group-set-selector (WIP)

- define and declare new component files

- import new component into app

- downgrade component

* chore: migrate group-set-selector

- convert AngularJS template syntax to Angular syntax
- finish implementing `group-set-selector.component.ts`
- remove redundant Coffeescript file

* chore: migrate group-set-selector

- finish defining Typescript logic
- update parent component to use Angular syntax for child component declaration

* chore: cleanup and documentation

- remove redundant Coffeescript file
- adjust frontend migration todo list

* chore: remove redundant file

- remove redundant Coffeescript file

* fix: attribute and event binding

- fix attribute and event binding syntax
- remove redundant template file

* refactor: two-way binding

- remove redundant directive attributes

- rename EventEmitter var name in child component

* Update README.md

* Update README.md

* refactor: replace Bootstrap components with Angular Material

- replace Bootstrap components with Angular Material components & directives

- refactor to use modern Angular structural directives

returnMarcco committed:
refactor: migrate/date service (#931)

* chore: date-service.coffee to TS

- add `date.service.ts` file
- add `date.service.spec.ts` unit test
- downgrade TS module such that it works within AngularJS

* chore: unlink date-service.coffee

- complete unlinking process of  `date-service.coffee` after migrating to Typescript
- manual linting/comment formatting

* chore: delete redundant coffeescript file

- delete `date-service.coffee` file after migration

* chore:  formatting

- format code with prettier

lachlan-robinson committed:
refactor: migrate progress dashboard (#927)

* chore: 9.x windows

* chore: create new component files

* chore: unlink old component + link new component

* chore: migrate functionality + update to tailwindcss and material UI

* chore: remove old component files

* chore: responsive columns for visualisations

* Update package-lock.json

* Update package.json

* Update package.json

* Update package-lock.json

* chore: update styling to better match original

* chore: add custom breakpoint for responsive change

* chore: tweak breakpoint for better visualisation visibility

* chore: update tutor view functionality + tweak event emitter name

---------

Co-authored-by: Boink <40929320+b0ink@users.noreply.github.com>

lachlan-robinson committed:
fix: burndown chart visualisation (#942)

* fix: update burndown data to render series of different lengths

* chore: cleanup imports

* chore: remove leftover testing comment from previous contributor

b0ink committed:
Merge pull request #929 from zara1504/remove/media-service

Remove Media Service

lachlan-robinson committed:
chore: resolve merge conflicts

lachlan-robinson committed:
chore: resolve futher conflicts

lachlan-robinson committed:
chore: resolve conflicts

lachlan-robinson committed:
chore: exlude packge-lock

lachlan-robinson committed:
chore: exclude package

b0ink committed:
chore: delete migrated progress dashboard html
